package/MDAnalysis/lib/util.py

```diff
@@ -1685,6 +1686,29 @@ def blocks_of(a, n, m):
 
     return np.lib.stride_tricks.as_strided(a, new_shape, new_strides)
 
+
+def group_consecutive_integers(arr):
+    """Split a sorted array of integers into a list of consecutive sequences.
+
+    Parameters
+    ----------
+    arr: :class:`numpy.ndarray`
+
+    Returns
+    -------
+    list of :class:`numpy.ndarray`
+
+    Examples
+    >>> arr = np.array([ 2,  3,  4,  7,  8,  9, 10, 11, 15, 16])
+    >>> group_consecutive_integers(arr)
+    [array([2, 3, 4]), array([ 7,  8,  9, 10, 11]), array([15, 16])]
+    """
+    mask = np.ones(len(arr)+1, dtype=bool)
+    mask[1:-1] = (arr[1:]-arr[:-1]) > 1
```

Comment:
So a weird numpy trick that might be better here....

```python
np.r_[True, (arr[1:] - arr[:-1]) > 1, True]
```

From what I can understand, `np.r_` is a weird hack for concatenating arrays

Reply:
Since there's already `np.ediff1d` I like using that:

```python
np.array_split(arr, np.where(np.ediff1d(arr) != 1)[0])
```

Reply:
I bet there is even cleaner with some careful thought..